Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -289,7 +295,7 @@ namespace AzToolsFramework
                 domOperation.GetType() == AZ::Dom::PatchOperation::Type::Replace)
             {
                 const auto childIterator = m_domOrderedChildren.begin() + childIndex;
-                delete *childIterator; // deleting the widget also automatically removes it from the layout
```

Comment:
I just fixed this in my new PR; the issue was that I wasn't taking the PropertyWidgets out of m_domOrderedChildren. I think you can revert your change here.